cuckoo/root_node.py

```diff
@@ -225,3 +212,33 @@ def _orjson_default(obj):
             return obj.dict()
         else:
             return jsonable_encoder(obj=obj)
+
+
+class IterByteIO(BytesIO):
```

Comment:
this few lines of code probably took the most time for this feature. purpose: take in a generator and convert it to a buffered byte reader

- for my initial testing, I used a `TextIO` and that was rather simple. although `ijson` works with both, byte and str, you will get a deprecation warning when using strings and also performance is worse. I just have to switch to bytes was my thinking ... ERROR
- there are no standard libs (as far as I could find them) that do this reliably.
- I ended up passing a `MagicMock` to `ijson.items()` to see what that C code is actually calling and found that is calling the rather unusual `readinto` on top of `read`
- implementation-wise I think that the `itertools.islice` method (suggested in one SO post) is performing better to chop off chunks of a generator rather than some `while(True): chunk + next(iter)` nonsense suggested by most SO posts